capstone/client/src/components/StoresPage.js

```diff
@@ -86,14 +187,67 @@ class StorePage extends Component {
     const overviewCards = (this.state.originalStores.length === 0) ? <CircularProgress id="stores-loading" color="action" /> : 
     <StoreOverviewCards stores={this.state.stores} numItems={this.state.items.length}/>;
 
-  const method = (this.state.method === "location") ? <h5>Calculated from tracked location</h5> : <h5>Calculated from Zip Code: {this.state.zipCode}</h5>
+    const method = (this.state.method === "location") ? <h5>Calculated from tracked location</h5> : <h5>Calculated from Zip Code: {this.state.zipCode}</h5>
+  
+    const shareButton = (this.state.originalStores.length === 0) ? null :
+        (<Button id="back-button" variant="contained" color="primary" onClick={this.onShare}>
+           Share Stores <MailOutlineIcon color='white' />
+         </Button>)
 
     return(
       <div id="stores-page-container">
         <h1>Store Recommendations</h1>
         {method}
         <StoresProvider>
-        <Button id="back-button" onClick={this.goBack} color="primary" variant="contained">Back To List</Button>
+        <Grid item>
+            <Button id="back-button" onClick={this.goBack} color="primary" variant="contained"><ArrowBackIcon color='white' />Back To List</Button>
+            {shareButton}
+        </Grid>
+        <Grid item>
+        <Dialog open={this.state.shareDialog.display} onClose={this.handleDialogCancel} aria-labelledby="form-dialog-title">
+          <DialogTitle id="form-dialog-title">Share Results</DialogTitle>
+          <DialogContent>
+            <DialogContentText>
+              To share your results with a friend, type in their email here.
```

Comment:
nit: Maybe instead of sharing it with a "friend" (because they could also be sending it to themselves), you could say something like: "Enter address to email results." OR "Send results to the email address below: " OR "Share results to: "